### PR TITLE
Create tokenized_dataloader.py

### PR DESCRIPTION
**Pull request by RileyWC**
created a dataloader for the tokenized data. 

The only inefficiency is that each tokenized sentence is padded to the max length of the entire dataset, not of the batch. i wasn't sure how to get it the other way.

### CODE REVIEW

File: src/tokenized_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        padding_length = self.max_length - len(tokens)
+        tokens.extend(['<pad>'] * padding_length)  # Padding with '<pad>'
+
+        # Return the input_ids, attention_mask, and label as a tuple
+        return tokens, label
+
+# Example usage
+dataset = SarcasmDataset(json_file="test_tokenized.json")
+
+dataloader = DataLoader(dataset, batch_size=32, shuffle=True)
```
Comment (tsunami935) [inline src/tokenized_dataloader.py]:
```suggestion
if __name__ == "__main__":
    # Example usage
    dataset = SarcasmDataset(json_file="test_tokenized.json")
    
    dataloader = DataLoader(dataset, batch_size=32, shuffle=True)
```

File: src/tokenized_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        
+        # Pad the tokens to the max length if needed
+        padding_length = self.max_length - len(tokens)
+        tokens.extend(['<pad>'] * padding_length)  # Padding with '<pad>'
+
+        # Return the input_ids, attention_mask, and label as a tuple
+        return tokens, label
+
+# Example usage
+dataset = SarcasmDataset(json_file="test_tokenized.json")
```
Comment (tsunami935) [inline src/tokenized_dataloader.py]: This line crashes btw.

File: src/tokenized_dataloader.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        item = self.data[idx]
+        label = torch.tensor(item['is_sarcastic'])
+        tokens = item['tokens']
+        
+        # crop for max length (shouldnt be needed but just in case)
+        tokens = tokens[:self.max_length]
+        
+        # Pad the tokens to the max length if needed
+        padding_length = self.max_length - len(tokens)
+        tokens.extend(['<pad>'] * padding_length)  # Padding with '<pad>'
```
Comment (tsunami935) [inline src/tokenized_dataloader.py]: I would suggest padding with the end of string token https://github.com/tsunami935/nlp-sarcasm-detection/blob/f3f8152d431769097ae87758c16c797499a09d86/src/common.py#L2.
"\<pad\>" token is not in the vocabulary.

File: src/tokenized_dataloader.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import nltk
+from nltk.tokenize import word_tokenize
+
+class SarcasmDataset(Dataset):
+    def __init__(self, json_file):
+        # Load data from the JSON file
+        self.data = []
+        with open(json_file, 'r') as f:
+            for line in f:
+                self.data.append(json.loads(line.strip()))
```
Comment (tsunami935) [inline src/tokenized_dataloader.py:15]: This is very slow.

File: src/tokenized_dataloader.py (hunk 3 above)
Comment (tsunami935) [reply src/tokenized_dataloader.py]: Also `extend` is slower than concatenation.

**Review by tsunami935 — APPROVED** [review]
